TrueCraft.Client/Rendering/Blocks/SlabRenderer.cs

```diff
@@ -0,0 +1,175 @@
+        {
+            switch (type)
+            {
+                case SlabType.Stone:
+                    return new Vector2(6, 0);
+
+                case SlabType.Sandstone:
+                    return new Vector2(0, 13);
+
+                case SlabType.Wood:
+                    return new Vector2(4, 0);
+
+                case SlabType.Cobblestone:
+                    return new Vector2(1, 0);
+
+                default:
+                    return Vector2.Zero;
+            }
+        }
+
+        protected virtual Vector2 GetBottomTexture(SlabType type)
+        {
+            switch (type)
+            {
+                case SlabType.Stone:
+                    return new Vector2(6, 0);
+
+                case SlabType.Sandstone:
+                    return new Vector2(0, 14);
+
+                case SlabType.Wood:
+                    return new Vector2(4, 0);
+
+                case SlabType.Cobblestone:
+                    return new Vector2(1, 0);
+
+                default:
+                    return Vector2.Zero;
+            }
+        }
+
+        protected virtual Vector2[] GetTextureMap(SlabType type, bool isDouble)
+        {
+            var sideTexture = GetSideTexture(type);
+            var topTexture = GetTopTexture(type);
+            var bottomTexture = GetBottomTexture(type);
+
+            var result = new Vector2[]
+            {
+                // Positive Z
+                sideTexture + Vector2.UnitX + ((isDouble) ? Vector2.UnitY : Vector2.UnitY / 2),
+                sideTexture + ((isDouble) ? Vector2.UnitY : Vector2.UnitY / 2),
+                sideTexture,
+                sideTexture + Vector2.UnitX,
+                // Negative Z
+                sideTexture + Vector2.UnitX + ((isDouble) ? Vector2.UnitY : Vector2.UnitY / 2),
+                sideTexture + ((isDouble) ? Vector2.UnitY : Vector2.UnitY / 2),
+                sideTexture,
+                sideTexture + Vector2.UnitX,
+                // Positive X
+                sideTexture + Vector2.UnitX + ((isDouble) ? Vector2.UnitY : Vector2.UnitY / 2),
+                sideTexture + ((isDouble) ? Vector2.UnitY : Vector2.UnitY / 2),
+                sideTexture,
+                sideTexture + Vector2.UnitX,
+                // Negative X
+                sideTexture + Vector2.UnitX + ((isDouble) ? Vector2.UnitY : Vector2.UnitY / 2),
+                sideTexture + ((isDouble) ? Vector2.UnitY : Vector2.UnitY / 2),
+                sideTexture,
+                sideTexture + Vector2.UnitX,
+                // Positive Y
+                topTexture + Vector2.UnitX + Vector2.UnitY,
+                topTexture + Vector2.UnitY,
+                topTexture,
+                topTexture + Vector2.UnitX,
+                // Negative Y
+                bottomTexture + Vector2.UnitX + Vector2.UnitY,
+                bottomTexture + Vector2.UnitY,
+                bottomTexture,
+                bottomTexture + Vector2.UnitX
+            };
```

Comment:
We need to generate texture arrays once and reuse them for all rendered blocks. See how logs do it, for example. It's very wasteful to allocate a new array for every block rendered.